packages/node/src/lib/ReadMe.ts

```diff
@@ -0,0 +1,212 @@
+          console.error(e.data);
+        } else {
+          console.error('Error calling ReadMe API. Contact support@readme.io for help!');
+          console.error(e.data);
+        }
+        // Don't want to cause an error in their API
+        return next();
+      }
+    }
+
+    if (req.path === '/readme-webhook' && req.method === 'POST' && !options.disableWebhook) {
+      try {
+        verifyWebhook(req.body, req.headers['readme-signature'] as string, readmeProjectData.jwtSecret as string);
+        const user = await userFunction(req, getUser);
+        return res.send(user);
+      } catch (e) {
+        return res.status(400).json({ error: (e as Error).message });
+      }
+    } else if (req.path === '/readme-setup' && options.development) {
+      const setupHtml = buildSetupView({
+        baseUrl,
+        subdomain: readmeProjectData.subdomain as string,
+        stableVersion: readmeVersionData?.find(version => version.is_stable)?.version || '1.0',
+        readmeAPIKey,
+        disableMetrics: options.disableMetrics,
+        disableWebhook: options.disableWebhook,
+      });
+      return res.send(setupHtml);
+    } else if (req.path === '/webhook-test' && options.development) {
+      const email = req.query.email as string;
+      try {
+        const webhookData = await testVerifyWebhook(baseUrl, email, readmeProjectData.jwtSecret as string);
+        return res.json({ ...webhookData });
+      } catch (e) {
+        return res.status(400).json({ error: (e as Error).message });
+      }
+    }
+
+    try {
+      const user = await userFunction(req, getUser);
+      if (!user || !Object.keys(user).length || options.disableMetrics) return next();
+
+      const groupId = getGroupIdByApiKey(user, requestAPIKey);
+      if (!groupId) {
+        console.error(
+          usingManualAPIKey
+            ? 'The API key you passed in via `manualAPIKey` could not be found in the user object you provided.'
+            : 'Could not automatically find an API key in the request. You should pass the API key via `manualAPIKey` in the `getUser` function. Learn more here: https://docs.readme.com/main/docs/unified-snippet-docs#/getuserbyapikey',
+        );
+        return next();
+      }
+
+      log(
+        readmeAPIKey,
+        req,
+        res,
+        {
+          apiKey: groupId,
+          label: user.label ? user.label : user.name,
+          email: user.email,
+        },
+        options,
+      );
+    } catch (e) {
+      return next();
+    }
+    return next();
+  };
+};
+
+/**
+ * This method configures the API Key for your ReadMe project and returns middleware that will
+ * send supplied API requests to ReadMe Metrics and configure the ReadMe webhook.
+ *
+ * @see {@link https://readme.com/metrics}
+ * @see {@link https://docs.readme.com/main/docs/unified-snippet-docs}
+ * @param readmeApiKey The API key for your ReadMe project. This ensures your requests end up in
```

Comment:
this should correspond to the function argument:

```suggestion
 * @param key The API key for your ReadMe project. This ensures your requests end up in
```